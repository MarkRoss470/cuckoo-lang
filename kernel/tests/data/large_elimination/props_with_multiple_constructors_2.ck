data False : Prop where

data SomeTy : Prop where
  | c1 : False -> SomeTy
  | c2 : False -> SomeTy

-- Even though neither of `SomeTy`'s constructors are actually inhabited, `SomeTy` is not large eliminating.
-- This `Prop` judgement is wrong, but never reached because the level error is encountered first
def some_ty_rec : Prop := SomeTy.rec.{1}