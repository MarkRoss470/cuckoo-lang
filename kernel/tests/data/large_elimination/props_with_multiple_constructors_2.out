ERROR at tests/data/large_elimination/props_with_multiple_constructors_2.ck 9:27-9:41: SomeTy.rec takes 0 level argument(s), but 1 were provided.